### PR TITLE
Hiding sensitive Information

### PR DESCRIPTION
**Pull request by sadakchap**
**Issue**
    https://github.com/E-Angadi/AngadiWeb/issues/8

**Solution** -
    Used environment variable & firebase function config to get algolia keys in frontend and backend respectively .

### CODE REVIEW

**Comment by sadakchap** [comment]
@tbharathchandra , I just noticed that `functions/config.js` we are only using `algolia.index_name` from this file, rest of the variables will come from firebase. Is this holding sensitive info or we can just leave it out there ?
Please, do look at the updated readme and changed code, and let me know if you need any further changes.

**Comment by tbharathchandra** [comment]
Hey @sadakchap, Sorry for this late response. I was bit busy switching between college and my work last week.



I've reviewed your changes and realized that **Firebase keys** are not in the env file. Hiding these keys are also important. I should have conveyed it clearly to you through the issue description. Sorry for that.

Can you make required changes and update the readme. This time I'll try to review as fastly as possible. ✌️✌️✌️

**Comment by sadakchap** [comment]
@tbharathchandra , thanks for reviewing it. I have now made following changes for firebase configs also.

But, while making changes I realized that in early commits I had made a change to get algolia index name key from firebase deployed variable although, we haven't deployed that variable to firebase as per our current readme. Trying to read it like this  `const ALGOLIA_INDEX_NAME = functions.config().algolia.index_name;` may give empty or undefined instead of `products` which can break the functionality.

Please let me know that, whether we want to deploy that variable also or should I change it back to get its value from local config file ( `/function/config`).

**Comment by tbharathchandra** [comment]
Index name is a not critical in terms of security and will not be changed more often. But is better to deploy it as a env variable. As it will be together with other algolia related configs. So, you should add this in readme.

**Comment by sadakchap** [comment]
I have added algolia index name as env variable in dummy `.env.local.txt` file. for readme, I have just mentioned the variable there, as will not be changing it, imho.

And now, all of config variables that were in `functions/config.js` are now deployed. I think we can remove this file since we are not using it.

File: README.md [hunk 1]
```diff
@@ -128,35 +128,35 @@ This is a detailed step by step instruction guide to setup your own e-commerce s
...
+5. Hit on the firestore tab in the left drawer in the firebase console and create the firestore database with security rules in test mode. Follow this [guide](https://firebase.google.com/docs/firestore/quickstart#create).
 
-6. Click on the web icon in project overview and create a web project with hosting enabled. and copy the Firebase SDK snippet in config mode from project settings. Replace these configs with the dummy ones below in `src/config/firebaseConfig.js`
+6. Similarly, setup Authentication with Email/Password and Anonymous mode enabled. Also setup functions and storage with no changes made.
+
+7. Click on the web icon in project overview and create a web project with hosting enabled. and copy the Firebase SDK snippet in config mode from project settings. Copy & paste each of these configs with the dummy ones present in `.env.local`.
    ```JS
     export var firebaseConfig = {
```
Comment (tbharathchandra) [inline README.md]: I think this config variable is irrelevant now, 

can you replace this config variable with the below chunk in the readme
 
```
REACT_APP_FIREBASE_API_KEY=your_firebase_apikey
REACT_APP_FIREBASE_AUTH_DOMAIN=your_firebase_authDomain
REACT_APP_FIREBASE_DATABASE_URL=your_firebase_databaseUrl
REACT_APP_FIREBASE_PROJECT_ID=your_firebase_projectId
REACT_APP_FIREBASE_STORAGE_BUCKET=your_firebase_storageBucket
REACT_APP_FIREBASE_MESSAGING_SENDER_ID=your_firebase_messagingSenderId
REACT_APP_FIREBASE_APP_ID=your_firebase_appId
```

**Review by tbharathchandra — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sadakchap) [reply README.md]: Sure thing, I have updated the readme as requested. Please take a look :)

**Comment by tbharathchandra** [comment]
@sadakchap, I think we don't need `functions/config.js` anymore, and have you tried deploying these functions to firebase with eslint check?

**Comment by sadakchap** [comment]
> @sadakchap, I think we don't need `functions/config.js` anymore, and have you tried deploying these functions to firebase with eslint check?

Right, we don't need `functions/config` anymore, so I have removed it and I haven't tried deploying functions with eslint check.
Now, when I did ran it,

image

which refers to this `functions/search.js`
image

Can this be its possible solution 
```JS

        return index
      })
      .saveObjects(arr)
      .then((context) => {
        return response.status(200).send(context);
      })
      .catch((err) => response.status(402).send(err));
  }
```

Let me know about it, so I make changes further.

**Comment by tbharathchandra** [comment]
Great, don't make that change in this PR. Anyway that is a warning. I'll review this branch once again and merge it ✌️

**Comment by tbharathchandra** [comment]
Push the changes

**Comment by sadakchap** [comment]
> Push the changes

changes with eslint warning fixes, right?

**Comment by tbharathchandra** [comment]
No, don't include those warning fixes in this PR

**Comment by sadakchap** [comment]
> No, don't include those warning fixes in this PR

Ok, I will just remove `functions/config.js` file :)

**Review by tbharathchandra — APPROVED** [review]

**Comment by sadakchap** [comment]
Thanks!!!

**Comment by tbharathchandra** [comment]
@sadakchap Thank you for your contributions. Please checkout other issues and hoping you will continue your contributions for Angadi Web.

**Pull request merged**